HW16/CoffeeShop.cpp

```diff
@@ -0,0 +1,111 @@
+#include <string>
+#include "Coffee.h"
+#include <vector>
+#include <iostream>
+
+class CoffeeShop
+{
+public:
+
+	std::vector<bool*> m_Tables;
+	std::vector<Coffee*> m_Orders;
+
+	CoffeeShop(int TABLES_COUNT, std::string m_name)
+	{
+		this->TABLES_COUNT = TABLES_COUNT;
+
+        this->m_Tables.resize(TABLES_COUNT);
+		for (int i = 0; i < TABLES_COUNT; i++)
+			m_Tables[i] = new bool(false);
+
+		this->m_Orders.resize(TABLES_COUNT);
+		for (int i = 0; i < TABLES_COUNT; i++)
+			m_Orders[i] = nullptr;
+
+		name = m_name;
+	}
+
+	std::string getName() { return name; };
+
+	int takePlace()
+	{
+		for (int i = 0; i < TABLES_COUNT; i++)
+			if (!(*m_Tables[i]))
+				return i;
+
+		return -1;
+	}
+
+	void order(int tableNumber)
+	{
+		int result;
+
+		while (true)
+		{
+			std::cout << "Place your order" << std::endl;
+			std::cout << "0 - Espresso" << std::endl;
+			std::cout << "1 - Cappuccino" << std::endl;
+			std::cout << "2 - Latte" << std::endl;
+			std::cout << "3 - Americano" << std::endl;
+			std::cout << "-1 to exit" << std::endl;
+			std::cout << "Your choice:";
+
+			std::cin >> result;
+
+			if (result == -1)
+				return;
+
+			if (result >= 0 && result <= 3)
+				break;
+		}
+
+		delete m_Tables[tableNumber];
+		m_Tables[tableNumber] = new bool(true);
```

Comment:
You should validate that tableNumber is not already occupied before reserving it once again.